### PR TITLE
Add rudimentary trace exporting and trace points to Machine ID

### LINKED ISSUE #13330: Add tracing to `tbot`
Following the example of `tsh`, introduce tracing to `tbot`. The spans should be submitted via the tracing endpoint on the auth server.

This will give us some visibility into the performance of `tbot` within live environments, and allow us to better understand how we may be able to optimise the renewal loop. It will also help with tracking down issues with `tbot` quicker.

For now, we should only run this when `tbot start` has been executed. There should be one root span for each renewal loop, and we should aim to ensure that each of the API client methods we call within the renewal loop are instrumented.

### PR DESCRIPTION
**Pull request by strideynet**
A rudimentary form of tracing intended for debugging and performance analysis. This is currently hidden as it does not work nicely when exporting to Teleport, but works fine if exporting to a local instance of jaeger.

Closes https://github.com/gravitational/teleport/issues/13330

https://github.com/gravitational/teleport/issues/35829 blocks our ability to ship traces to Teleport meaningfully.

Example of Traces showing in a local jaeger (`--trace --trace-exporter=grpc://localhost:4317`):

image

Some interesting learnings from this already - hopefully we can continue to build on this.

### CODE REVIEW

File: lib/tbot/bot_identity.go [hunk 1]
```diff
@@ -47,6 +47,8 @@ func (b *Bot) renewBotIdentityLoop(
 	ctx context.Context,
 	reloadChan <-chan struct{},
 ) error {
+	ctx, span := tracer.Start(ctx, "Bot.renewBotIdentityLoop")
```
[the remark this answers is not in the document]
Comment (strideynet) [reply lib/tbot/bot_identity.go]: Addressed by [dc20abb](https://github.com/gravitational/teleport/pull/35833/commits/dc20abbcbbe2c7878535ca3120b27e271be59c83)

**Review by rosstimothy — APPROVED** [review]
🎉

**Review by mdwn — APPROVED** [review]
I love this -- I'd love to see more integration with OTEL throughout Teleport (if we don't already have it), but this will do wonders for us.

**Comment by strideynet** [comment]
> I'd love to see more integration with OTEL throughout Teleport

@rosstimothy has done a fantastic job of putting the foundation in and I believe we collect traces for all new cloud tenants. At this point we just need to convince more engineers to get in the swing of adding spans across Teleport >:3

**Pull request merged**